include/internal/thread_once.h

```diff
@@ -30,6 +30,14 @@
     }                                           \
     static int init(void)
 
+#define DEFINE_RUN_ONCE_STATIC_ALT(initalt, init) \
+    static int initalt(void);                     \
+    static void initalt##_ossl_(void)             \
+    {                                             \
+        init##_ossl_ret_ = initalt();             \
```

Comment:
Shouldn't there be a `static int init##_ossl_ret_ = 0;` somewhere in this macro definition?

Reply:
No. That's kind of the point of introducing this variant. This fixes a bug in the RUN_ONCE code. Previously we had the concept of declaring more than one variant of an initialiser function where all the variants all shared the same CRYPTO_ONCE object - so only one variant would ultimately get run (dependant on the options you choose at run time). Unfortunately each variant had their own `init##_ossl_ret` variable. So if you called `RUN_ONCE(myaltinit)` followed later by `RUN_ONCE(myinit)`, the later `RUN_ONCE` call always returned 0 - because the return value for that alternative had never been set. This is incorrect behaviour. I've tried to explain it in the comments. Please take a look.

Reply:
Agreed. The RUN_ONCE changes are correct.  Have not looked at the rest of this PR.